### PR TITLE
surface the cause of the json rules engine

### PR DESCRIPTION
**Pull request by punkle**
## Hey, I just made a Pull Request!

surface the cause of the json rules engine

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts [hunk 1]
```diff
@@ -141,7 +141,9 @@ export class JsonRulesEngineFactChecker
       );
     } catch (e) {
       if (isError(e)) {
-        throw new Error(`Failed to run rules engine, ${e.message}`);
+        throw new Error(`Failed to run rules engine, ${e.message}`, {
+          cause: e,
+        });
```
Comment (freben) [inline plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts:146]: `e` should just be the plain second argument directly, right? Not wrapped in an object. Or am I misunderstanding that?

Comment (punkle) [reply plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts:146]: I am not sure what the right thing to do is. I had been following an example like this https://nodejs.org/api/errors.html#errorcause

Comment (punkle) [reply plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts:146]: I am happy with either approach so long as I can retrieve the exception cause running the rules engine.

Comment (benjdlambert) [reply plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts:146]: Let's go with the one that's in the docs :shrug:

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**